### ISSUE TITLE
chromosomes should return String array or make GenomicRanges accept Symbol array

### ISSUE THREAD

**Issue by Marlin-Na** [phaverty/GenomicVectors.jl#26]
I think GenomicRanges perhaps should accept `Array{Symbol,1}` as chromosomes as well since `chromosomes` function is returning `Array{Symbol,1}`. Alternatively, `chromosomes` should return a String Array (which I prefer, since strings might be easier to work with).

Currently, I have to use `map` to replicate a GenomicRanges:
```
GenomicRanges(map(String, chromosomes(gr)), starts(gr), ends(gr), chr_info(gr))
```
Ideally, this should work:
```
GenomicRanges(chromosomes(gr), starts(gr), ends(gr), chr_info(gr))
```
Anyway, this should be a minor issue on UX.

**Comment by phaverty** [comment]
I thought about this a bunch when I did my last big batch of work on this project. Symbols seem like the right thing to use internally because I want them to be immutable labels. I've even played with making them part of the type definition by making the chromosomes slot a NamedTuple. I probably want both Symbol and String to work in the user API. I should experiment with this more now that Strings are more efficient (in 1.5+).

**Comment by Marlin-Na** [comment]
Thanks @phaverty . I don't have much to comment about the Symbol vs String for internal use. But your idea of making them as part of type definition is interesting. Even chromosome lengths could be part of the type definition, so that it could be used to parameterize scalar integer as "genopos" (mentioned in #25 ). But such complex type probably would slow down JIT compiling. Also it won't work with megagenome, which might contain thousands of contigs.

**Comment by phaverty** [comment]
I did a bit of work to get GenomicVectors updated for some recent updates in Julia and dependency packages. I'll be able to have a look at this now.

**Comment by phaverty** [comment]
Fixed by 47078eca3724cedd9aada430697427d7408d9397
